### PR TITLE
[MO] add reinterp_shape for StridedSlice

### PR DESCRIPTION
**Pull request by pavel-esir**
**Root cause analysis**:  For `NHWC` nets Squeezing\unsqueezing should be done in the original FW layout regardless of operation with which they are expressed Reshape, StridedSlice (SS), Squeeze, Unsqueeze. 

We have a transformation that replaces SS with shring\new_axis masks by Squeeze\Unsqueeze. And for Squeeze\Unsqueese there is a `reinterp_shape` attribute which takes care of inserting Transposes to restore original FW `NHWC` layout. 

But it is triggered only if squeezing/unsqueezing is the only thing that SS does. This is not always the case, e.g. PixelLink has SS which slices data and at the same time Squeezes. 

**Solution**: Remove solution #6693. Instead add reinterp_shape attribute to StridedSlice.

Ticket: CVS-73336

Code:
* [x]  Comments
* [x]  Code style (PEP8)
* [x]  Transformation generates reshape-able IR
* [x]  Transformation preserves original framework node names

Validation:
* [x] Unit tests
* [x]  Framework operation tests: layer tests passed
* n/a  Transformation tests
* [x]  Model Optimizer IR Reader check

Documentation:
* n/a Supported frameworks operations list: n/a
* n/a Guide on how to convert the **public** model: n/a
*  n/a User guide update: n/a

### CODE REVIEW

**Review by jane-intel — APPROVED** [review]

**Comment by jane-intel** [comment]
@pavel-esir please merge upstream/master

**Review by achetver — APPROVED** [review]

**Comment by pavel-esir** [comment]
> @pavel-esir please merge upstream/master

Done. Corrected old unit-tests as well.

**Review by sadolini — APPROVED** [review]

**Review by vgavrilo — APPROVED** [review]

**Comment by rkazants** [comment]
@pavel-esir, please investigate performance degradation in a couple of models

**Review by rkazants — CHANGES_REQUESTED** [review]
please investigate performance degradation in a couple of models

**Comment by pavel-esir** [comment]
> please investigate performance degradation in a couple of models

@rkazants Looks like the reported performance changes are rather random deviations than real issues. I wrote more detailed report to email. Please review

File: tools/mo/openvino/tools/mo/middle/StridedSliceNormalizer.py [hunk 1]
```diff
@@ -113,6 +113,12 @@ def find_and_replace_pattern(self, graph: Graph):
             if node.is_in_port_connected(3):
```
Comment (rkazants) [inline tools/mo/openvino/tools/mo/middle/StridedSliceNormalizer.py:113]: Please check that attributes of this operation would not be adjusted to NCHW layout by this code above:
```
            PermuteAttrs.create_permute_attrs(node,
                                              attrs=[('begin_mask', 'input:0'),  # but indeed depends from slice_rank
                                                     ('end_mask', 'input:0'),
                                                     ('new_axis_mask', 'input:0'),
                                                     ('shrink_axis_mask', 'input:0'),
                                                     ('ellipsis_mask', 'input:0')])
```

Comment (rkazants) [reply tools/mo/openvino/tools/mo/middle/StridedSliceNormalizer.py:113]: It looks that permuted attrs are saved in additional node attribute but they won't applied since  node['nchw_layout'] = True

File: tools/mo/unit_tests/mo/middle/StridedSliceNormalizer_test.py [hunk 2]
```diff
@@ -1644,96 +1650,44 @@ def test_strided_slice_permute_2(
...
     def test_strided_slice_permute_7(
-            self,  # inp[0, 20], ends can be of any value
-            inp=(1, 35, 35, 3), ref_res=(3, 35),
-            begin=(0, 20), end=(1, 9999), strides=(1, 1), begin_mask=(0,), end_mask=(0,),
-            shrink_axis_mask=(1, 1), new_axis_mask=(0,), ellipsis_mask=(0,)
-    ):
-        self.run_permute_test(inp, ref_res, begin, end, strides,
-                              begin_mask, end_mask, shrink_axis_mask, new_axis_mask, ellipsis_mask)
-
-    def test_strided_slice_permute_8(
```
Comment (rkazants) [inline tools/mo/unit_tests/mo/middle/StridedSliceNormalizer_test.py:1690]: why did you remove a lot of tests? Please recover them if possible

**Review by rkazants — COMMENTED** [review]
Left one comment regarding tests. The remained code looks good.

File: tools/mo/unit_tests/mo/middle/StridedSliceNormalizer_test.py (hunk 2 above)
Comment (pavel-esir) [reply tools/mo/unit_tests/mo/middle/StridedSliceNormalizer_test.py:1690]: I removed them, because at the beginning I added them to check that attributes are changed accordingly with layout but now since for some cases (when srink/new axis are nonzero) attributes are not changed when with layout. 

So I removed them because they were red (expectedly) and corrected some others.

**Review by rkazants — APPROVED** [review]
